rfcs/supported/algorithms_histogram_cpu_backends/README.md

```diff
@@ -138,38 +138,96 @@ percentage of the output histogram may be occupied, even when considering dividi
 histogram-specific patterns in the implementation of a backend.
 
-This algorithm does however require that each parallel backend will add a
-`__enumerable_thread_local_storage<_StoredType>` struct which provides the following:
-* constructor which takes a variadic list of args to pass to the constructor of each thread's object
-* `get_for_current_thread()` returns reference to the current thread's stored object
-* `get_with_id(int i)` returns reference to the stored object for an index
-* `size()` returns number of stored objects
-
-In the TBB backend, this will use `enumerable_thread_specific` internally. For OpenMP, we implement our own similar
-thread local storage which will allocate and initialize the thread local storage at the first usage for each active
-thread, similar to TBB. The serial backend will merely create a single copy of the temporary object for use. The serial
-backend does not technically need any thread specific storage, but to avoid special casing for this serial backend, we
-use a single copy of histogram. In practice, our benchmarking reports little difference in performance between this
-implementation and the original, which directly accumulated to the output histogram.
-
-With this new structure we will use the following algorithm:
-
-1) Run a `parallel_for` pattern which performs a `histogram` on the input sequence where each thread accumulates into
-its own temporary histogram returned by `__enumerable_thread_local_storage`. The parallelism is divided on the input
-element axis, and we rely upon existing `parallel_for` to implement chunksize and thread composability.
-2) Run a second `parallel_for` over the `histogram` output sequence which accumulates all temporary copies of the
-histogram created within `__enumerable_thread_local_storage` into the output histogram sequence. The parallelism is
-divided on the histogram bin axis, and each chunk loops through all temporary histograms to accumulate into the
-output histogram.
+
+#### Enumerable Thread Local Storage feature
+This algorithm requires that each parallel backend add a "make" function with the following signature:
+```
+template <typename _ValueType, typename... _Args>
+/* unspecified enumerable thread local storage type */
+oneapi::dpl::__par_backend::__make_enumerable_tls(_Args&&)
+```
+returning an object which represents an enumerable thread local storage. The enumerable thread local storage object
+must provide the following interfaces:
+
+* `__ValueType& get_for_current_thread()`
+
+ Returns reference to the current thread's stored object. This function is
+ meant to be used within a parallel context, where each thread can access storage exclusive to its thread index. The
+ thread local storage should be created on first call of this function per thread. This avoids unnecessary overhead
+ for loops with fewer threads used than maximum.
```

Comment:
```suggestion
 Returns reference to the current thread's stored object. This function is meant to be used within a parallel context,
 with each thread exclusively accessing its part of the storage.
```
I dropped the implementation-specific part of the wording, as I believe it should better be separate from API description. It also applies to other suggestions below.